p2p/peermanager.go

```diff
@@ -1205,13 +1245,24 @@ func (p *peerInfo) Copy() peerInfo {
 // Score calculates a score for the peer. Higher-scored peers will be
 // preferred over lower scores.
 func (p *peerInfo) Score() PeerScore {
-	var score PeerScore
 	if p.FixedScore > 0 {
 		return p.FixedScore
 	}
 	if p.Persistent {
-		score += PeerScorePersistent
+		return PeerScorePersistent
 	}
+
+	var score PeerScore
+
+	switch {
+	case p.MutableScore+int64(score) > math.MaxUint8:
+		score = math.MaxUint8
+	case p.MutableScore+int64(score) < 0:
+		score = 0
+	default:
+		score += PeerScore(p.MutableScore)
```

Comment:
Maybe I'm reading this wrong, but why are we adding anything? Shouldin't this just be an if-else?

```
if p.MutableScore < 0 {
 return 0
}

return max(p.MutableScore, math.MaxUint8)
```